### PR TITLE
[T3A3][T14-A1]Yeo Kai Yan

### PR DESCRIPTION
**Pull request by kaiyanyeo**

### CODE REVIEW

**Comment by kaiyanyeo** [comment]
Ready for review

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -8,11 +8,16 @@
...
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String EXAMPLE = "123, some street, some unit, some postal code";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses should be in"
+            + "a/BLOCK, STREET, UNIT, POSTAL_CODE format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
     public final String value;
+    private Block block;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: recollect the discussion during the tutorial on why you should not retain the `value` anymore.

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -26,6 +31,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = address;
+
+        String[] addressDetails = address.split(",");        
+        this.block = new Block(addressDetails[0]);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: try to remove the magic numbers?

File: src/seedu/addressbook/data/person/Block.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package seedu.addressbook.data.person;
+
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: Please include a header comment to indicate what this class/method does?

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -26,6 +31,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
+        this.street = new Street(addressDetails[1]);
+        this.unit = new Unit(addressDetails[2]);
+        this.postalCode = new PostalCode(addressDetails[3]);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: - how do you check for the equality of the address?
- how do you generate the hashcode for use later?

File: src/seedu/addressbook/data/person/Unit.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package seedu.addressbook.data.person;
+
+public class Unit {
+    private final String value;
+    
+    public Unit(String unit){
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Unit.java]: need a space between the braces and the text; what do coding standards say?  ;-)

File: test/expected.txt [hunk 6]
```diff
@@ -85,57 +85,57 @@
...
 || Enter command: || [Command entered:  list]
-|| 	1. Adam Brown Phone: 111111 Email: adam@gmail.com Address: 111, alpha street Tags: 
+|| 	1. Adam Brown Phone: 111111 Email: adam@gmail.com Address: 111, alpha street, 02-338, 543210 Tags: 
 || 
 || 1 persons listed!
 || ===================================================
-|| Enter command: || [Command entered:  add Betsy Choo pp/222222 pe/benchoo@nus.edu.sg pa/222, beta street t/secretive]
-|| New person added: Betsy Choo Phone: (private) 222222 Email: (private) benchoo@nus.edu.sg Address: (private) 222, beta street Tags: [secretive]
+|| Enter command: || [Command entered:  add Betsy Choo pp/222222 pe/benchoo@nus.edu.sg pa/222, beta street, 08-987, 987654 t/secretive]
+|| New person added: Betsy Choo Phone: (private) 222222 Email: (private) benchoo@nus.edu.sg Address: (private) 222, beta street, 08-987, 987654 Tags: [secretive]
```
Comment (okkhoy) [inline test/expected.txt]: good to see you have updated userguide, test and in program help :+1:

File: test/java/seedu/addressbook/parser/ParserTest.java [hunk 7]
```diff
@@ -266,7 +266,7 @@ private static String convertPersonToAddCommandString(ReadOnlyPerson person) {
                 + person.getName().fullName
                 + (person.getPhone().isPrivate() ? " pp/" : " p/") + person.getPhone().value
                 + (person.getEmail().isPrivate() ? " pe/" : " e/") + person.getEmail().value
-                + (person.getAddress().isPrivate() ? " pa/" : " a/") + person.getAddress().value;
+                + (person.getAddress().isPrivate() ? " pa/" : " a/") + person.getAddress().toString();
```
Comment (okkhoy) [inline test/java/seedu/addressbook/parser/ParserTest.java]: how do you generate the string representation? now that i have suggested to remove the `value` variable, you need a way to generate the original address from the components.

**Comment by okkhoy** [comment]
@kaiyanyeo 
Some comments added. Please ack & close the PR after reading comments.

**Pull request closed without merging**